main/distributed-programming.md

```diff
@@ -1,47 +1,55 @@
-# Distributed programming in JavaScript
+# Agoric JavaScript Programming Extensions
 
-Agoric's platform is built mainly in JavaScript and enables authors to write secure smart contracts in JavaScript.
+Agoric's platform lets you write secure smart contracts in JavaScript. The platform itself is mainly written in JavaScript. However, we've made several Agoric-specific additions to general JavaScript programming that you should know about and understand before programming on the platform. Some are *concepts*, others are *Agoric library additions*, and some are at the *syntax level*.
 
-For the biggest part, all JavaScript knowledge is directly applicable to the Agoric platform. However, there are a couple of Agoric-specific additions at various layers that are better to understand before programming on the Agoric platform.
+Extensions covered in this document are:
+- **[Vats](#vats)**: Objects and functions In the same JavaScript vat can
+  communicate synchronously. Communication with objects outside the
+  vat can only be done asynchronously. 
 
-Some specificities are **concepts**, other are at the **syntax level**, others are **Agoric library additions**.
+- **Secure EcmaScript (SES)**: SES provides a secure platform for
+  executing programs. With SES, you can run code you don't completely trust,
+  without being vulnerable to bugs or bad intentions. It's a
+  standards-track extension to the JavaScript standard. 
 
+- **Realms and compartments**: JavaScript code runs in the context of
+  a *realm*, made up of the set of primordials (the standard
+  library's objects and functions) and a global object. In a web
+  browser, an iframe is a realm. In Node.js, a Node process is a
+  realm. A *compartment* is a separate execution environment within a realm.
 
-## Vats
+- **`harden()`**: A hardened object’s properties cannot be changed, so the only way to interact with a hardened object is through its methods. `harden()`is similar to `Object.freeze()` but
+more powerful. 
 
-A vat is a *unit of synchrony*. This means that within a JavaScript vat, objects
-and functions can communicate with one another synchronously. It also means that
-communication with objects outside the vat can be done, but it requires
-[managing asynchronous messages and responses](#Communicating-with-remote-objects-via-E).
+- **Remote object communication using `E`**: `E` is a local "bridge" function that lets you invoke methods on remote objects, whether in another vat, machine, or blockchain (for example).
+It takes a local representative (a *proxy*) for a remote object as an argument and sends messages to it using normal message-sending syntax. The local proxy forwards all messages to the remote object to deal with. If you are familiar with the `~` operator, `E` is a workaround for the JavaScript standard not yet supporting `~`.
```

Comment:
It's not the "\~" operator, it's "\~.", pronounced til-dot, by decree of a twitter poll.

Reply:
The JS standards proposal document uses the more grownup-sounding "wavy dot": https://github.com/tc39/proposal-wavy-dot . But all its friends use "tildot".